### ISSUE TITLE
[Semantics]: Textfield with `obscureText: true` is labeled as `password`

### ISSUE THREAD

**Issue by HarminderSinghSandhu** [flutter/flutter#91470]
I have a CVV field in app, talkback announce CVV label when there is  **obscureText** is false on Textfield but if it is true then it say it "password".

Expected: It should pronounce labelText regardless of  obscureText value.

**Comment by danagbemava-nc** [comment]
Hi @HarminderSinghSandhu  

Can you please provide your `flutter doctor -v` and `flutter run --verbose` logs, and a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example)?

Thank you

**Comment by HarminderSinghSandhu** [comment]
This is the video stating the issue

Actual: Talkback says "Editing password 7 character edit-box"
Expected: "Editing CVV 7 character edit-box"

Textfield label name should be announced instead of password. This case is specific to **obscureText**  

https://user-images.githubusercontent.com/39084531/136536929-43374631-607c-49f3-ad2a-1970b9d31f4c.mp4


-------------------------------------------------

**flutter doctor -v**

[√] Flutter (Channel stable, 2.5.2, on Microsoft Windows [Version 10.0.19042.1237], locale en-US)
    • Flutter version 2.5.2 at C:\Users\hsingh\Documents\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 3595343e20 (8 days ago), 2021-09-30 12:58:18 -0700
    • Engine revision 6ac856380f
    • Dart version 2.14.3

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\hsingh\AppData\Local\Android\Sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = C:\Users\hsingh\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.0)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31605.320
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] Connected device (4 available)
    • sdk gphone x86 (mobile) • emulator-5554 • android-x86    • Android R (API 29) (emulator)
    • Windows (desktop)       • windows       • windows-x64    • Microsoft Windows [Version 10.0.19042.1237]
    • Chrome (web)            • chrome        • web-javascript • Google Chrome 94.0.4606.71
    • Edge (web)              • edge          • web-javascript • Microsoft Edge 94.0.992.38

• No issues found!


**flutter run --verbose logs** 
[runVersoseLogs.txt](https://github.com/flutter/flutter/files/7310356/runVersoseLogs.txt)

**Comment by danagbemava-nc** [comment]
Hi @HarminderSinghSandhu, thanks for the video sample. 

Can you please provide a minimal reproducible code sample? 

Thank you

**Comment by danagbemava-nc** [comment]
Issue is reproducible on `stable` and `master`. 

I can only reproduce it with `talkback(android)` as `voiceover(iOS)` correctly identifies the label as `CVV`. 

`Talkback` however, always labels it as `password`. If `obscureText` is false, it reads label as `CVV` field, however if it is true, it always reads it as `password`

<details>
<summary>flutter doctor -v</summary>

```console
[✓] Flutter (Channel stable, 2.5.2, on macOS 11.5.1 20G80 darwin-arm, locale en-GB)
    • Flutter version 2.5.2 at /Users/nexus/dev/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 3595343e20 (8 days ago), 2021-09-30 12:58:18 -0700
    • Engine revision 6ac856380f
    • Dart version 2.14.3

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.60.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.27.0

[✓] Connected device (5 available)
    • M2007J20CG (mobile)    • 5dd3be00                             • android-arm64  • Android 11 (API 30)
    • Nexus’ IPhone (mobile) • 00008020-001875E83A38002E            • ios            • iOS 14.8 18H17
    • iPhone 12 (mobile)     • F2592D43-5FE3-46BB-8A2B-A38BA468DCE9 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 11.5.1 20G80 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 94.0.4606.71

• No issues found!
```

```console
[✓] Flutter (Channel master, 2.6.0-12.0.pre.268, on macOS 11.5.1 20G80 darwin-arm, locale en-GB)
    • Flutter version 2.6.0-12.0.pre.268 at /Users/nexus/dev/sdks/flutters
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b9d2177da0 (10 hours ago), 2021-10-07 21:11:07 -0700
    • Engine revision 8d64613c9b
    • Dart version 2.15.0 (build 2.15.0-178.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 12.5.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.0.0.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.60.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.27.0

[✓] Connected device (5 available)
    • M2007J20CG (mobile)    • 5dd3be00                             • android-arm64  • Android 11 (API 30)
    • Nexus’ IPhone (mobile) • 00008020-001875E83A38002E            • ios            • iOS 14.8 18H17
    • iPhone 12 (mobile)     • F2592D43-5FE3-46BB-8A2B-A38BA468DCE9 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 11.5.1 20G80 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 94.0.4606.71

! Doctor found issues in 1 category.
```
</details>

**Comment by justinmc** [comment]
I want to clarify that on Android it does still say the label in addition to "password".  So for a field with `obscureText: true` and `labelText: 'my label'`, Talkback says, "password, my label, edit box".

Looking through the Android docs, this might be the best we can do.  In the docs for [AccessibilityNodeInfo](https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo#setPassword(boolean)), there doesn't seem to be a way to indicate that the field is obscured but not a password.  Currently [we call setPassword](https://github.com/flutter/engine/blob/90424212a3d6839ca7552b84f6d1108042bd0d94/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L633) if the field is obscured.

If I create a native Android app, I don't think I can do better either.  There are other [inputType](https://developer.android.com/reference/android/widget/TextView#attr_android:inputType)s, but the closest seems to be "numberPassword", and it gets announced as "password" by Talkback.

I'm going to close this, but let me know if I'm wrong and there is a better way to do this and I'll take a look.
